decomposed/extensions/MDDE_LDM/Profile/Model/Methods/mdde_InitializeExportExtension.xml

```diff
@@ -32,57 +51,80 @@ Function GTLOutputHasErrors(GTLResult)
    End If
 End Function
 
-Sub processPackage(objModel, objPackage, attributeTemplateName, attributeExtendedAttributeName)
+Function processPackage(objModel, objPackage, attributeTemplateName, attributeExtendedAttributeName)
 
    &#39;Loop over Entities
-   Dim ent
+   Dim ent, success1, success2
+   processPackage = True
+   success1 = False
+   success2 = False
    For Each ent in objPackage.Entities
       If Not ent.IsShortCut Then
-         applyTemplateOnObject objModel, ent, &quot;mddex_Entity_Export_Template&quot;, &quot;mddex_Entity_Export_Content&quot;
-         applyTemplateOnObject objModel, ent, attributeTemplateName, attributeExtendedAttributeName            
+         success1 = applyTemplateOnObject (objModel, ent, &quot;mddex_Entity_Export_Template&quot;, &quot;mddex_Entity_Export_Content&quot;)
```

Comment:
I guess you are trying to apply the template using 2 different methods, and in the end you want to see whether one of them succeeded?

Since in the If one line 66 with an Or, you are only interested in one of them succeeding? Can't you then check after success1 whether it succeeded and only execute the next apply of it was false?

Reply:
Also within each For Each loop here the code looks very similar. Can't you move it into a Function and pass the stereotype name?

Reply:
The reason that there are two invocation is that two different templates are applied on these metaclasses, being the metaclass specific template and the one that is defined on named object.

Reply:
Then both should succeed right? The If should then have an And i.s.o. Or right?

Reply:
Refactored it to a separate function